diff --git a/ytypes/BUILD.bazel b/ytypes/BUILD.bazel
index 1234567..0000000 100644
--- a/ytypes/BUILD.bazel
+++ b/ytypes/BUILD.bazel
@@ -33,6 +33,8 @@ go_library(
         "//ygot",
         "@com_github_golang_glog//:go_default_library",
         "@com_github_kylelemons_godebug//pretty:go_default_library",
+        "@com_github_google_go_cmp//cmp",
+        "@com_github_openconfig_gnmi//value:go_default_library",
         "@com_github_openconfig_gnmi//proto/gnmi:go_default_library",
         "@com_github_openconfig_goyang//pkg/yang:go_default_library",
         "@org_golang_google_grpc//codes",
